torch/onnx/_internal/exporter/_core.py

```diff
@@ -434,43 +439,117 @@ def _handle_call_function_node_with_lowering(
-        raise _errors.DispatchError(
-            f"No ONNX function found for {node.target!r}. Failure message: {message}"
+    if node.target.__class__.__name__ == "CondOp":
+        # TODO: The condition should be replaced by something more relevant.
+        # Controlflow is handled in the exporter and not in torchlib
+        # as subgraph are only available in this code.
+        cond, true_graph, false_graph, graph_args = node.args
+        assert (
+            true_graph.name in local_functions
+        ), f"Unable to find {true_graph.name!r} in local_functions={local_functions}"
+        assert (
+            false_graph.name in local_functions
+        ), f"Unable to find {false_graph.name!r} in local_functions={local_functions}"
+        # builds local functions
+        val_args = tuple(n.meta["val"] for n in graph_args)
+        # Expected `mod` to be an instance of `torch.nn.Module`, got <class 'torch.fx.graph.Graph'>.
+        # But this scenario should be supported.
+        true_program = export(local_functions[true_graph.name], val_args, registry=registry)
+        false_program = export(local_functions[false_graph.name], val_args, registry=registry)
+        true_function = ir.Function(
+            LOCAL_FUNCTION_DOMAIN, true_graph.name, graph=true_program.model.graph, attributes={}
+        )
+        false_function = ir.Function(
+            LOCAL_FUNCTION_DOMAIN, false_graph.name, graph=false_program.model.graph, attributes={}
+        )
+        #model.functions[LOCAL_FUNCTION_DOMAIN, true_graph.name, ""] = true_function
+        #model.functions[LOCAL_FUNCTION_DOMAIN, false_graph.name, ""] = false_function
+        # source = node.all_input_nodes[0]
+        # source_outputs = node_name_to_values[source.name]
+        inputs = [node_name_to_values[n.name] for n in graph_args]
+        onnx_inputs = [i.name for i in inputs]
+        # TODO: needs more tests to handle multiple outputs.
+        # The fx.graph only contains one outputs. If there are multiple,
+        # this node will be followed by getitem nodes.
+        outputs = [node.name]
+        onnx_outputs = outputs
+
+        def mkv(name):
+            value_info_proto = onnx.ValueInfoProto()
+            value_info_proto.name = name
+            return value_info_proto
+
+        if_node = oh.make_node(
+            "If",
+            [cond.name],
+            outputs,
+            then_branch=oh.make_graph(
+                [oh.make_node(true_graph.name, onnx_inputs, onnx_outputs, domain=LOCAL_FUNCTION_DOMAIN)],
+                true_graph.name,
+                [],
+                [mkv(o) for o in outputs],
+            ),
+            else_branch=oh.make_graph(
+                [oh.make_node(false_graph.name, onnx_inputs, onnx_outputs, domain=LOCAL_FUNCTION_DOMAIN)],
+                false_graph.name,
+                [],
+                [mkv(o) for o in outputs],
+            ),
+        )
```

Comment:
Avoid any proto operations.

Reply:
I don't really know how to write this with the ir. The documentation does not have many examples.

Reply:
I was hoping for some help here.

Reply:
I replaced with onnx api with onnxscript api.